schematic/visualization/attributes_explorer.py

```diff
@@ -59,17 +59,55 @@ def parse_attributes(self, save_file=True):
                 for the provided data model for all components in the data model. 
                 Dataframe is saved locally as a csv if save_file == True, or returned if
                 save_file == False. 
-        Raises:
-            ValueError:
-                If unable hits an error while attempting to get conditional requirements. 
-                This error is likely to be found if there is a mismatch in naming.
+                
         '''
         # get all components
         component_dg = self.sg.se.get_digraph_by_edge_type('requiresComponent')
         components = component_dg.nodes()
+        
         # For each data type to be loaded gather all attribtes the user would
         # have to provide.
+        return self._parse_attributes(components, save_file)
+    
+    def parse_component_attributes(self, component=None, save_file=True, include_index=True):
+        '''
+        Args: save_file (bool):
+                True: merged_df is saved locally to output_path.
+                False: merged_df is returned.
+
+        Returns:
+            merged_df (pd.DataFrame): dataframe containing data relating to attributes
+                for the provided data model for the specified component in the data model. 
+                Dataframe is saved locally as a csv if save_file == True, or returned if
+                save_file == False. 
+        '''        
 
+        if not component:
+            raise ValueError("You must provide a component to visualize.")
+        else:
+            return self._parse_attributes([component], save_file, include_index)
+
+    def _parse_attributes(self, components, save_file=True, include_index=True):
```

Comment:
Could you add tests for this function: `_parase_attributes?`

Reply:
@linglp this function is covered by tests `test_visualize_component` and `test_visualize_attributes` that are in `test_api.py::TestSchemaVisualization`

Reply:
@GiaJordan But these two tests only ensure that response code are 200? I think we should still add tests more specifically for ensuring that parse attributes return expected output?

Reply:
Also, the content of `_parse_attributes` is was just pulled from `parse_attributes` into its own function so that it could be called from another function as well, with the added flag